shared/util/codeql/util/FilePath.qll

```diff
@@ -0,0 +1,74 @@
+/** Provides a utility for normalizing filepaths. */
```

Comment:
I think it would be worth being clear in the docs about what this does and doesn't do; i.e. that it normalises `..` as much as possible, but doesn't resolve symlinks, normalise case, or absolutify paths.

Reply:
And that it treats only `/` as a path separator.